src/cobald/decorator/logger.py

```diff
@@ -10,6 +12,34 @@
 )
 
 
+class _WarnValue(NamedTuple):
+    value: Any
+    warning: Warning
+
+
+class _WarnMap(dict):
+    r"""Map raising ``warnings`` for specific keys pointing to ``_WarnValue``\ s"""
+
+    def __getitem__(self, item):
+        value = super().__getitem__(item)
+        if isinstance(value, _WarnValue):
+            warnings.warn(value.warning, stacklevel=2)
+            return value.value
+        else:
+            return value
+
+
+_DEPRECATION_MAP = _WarnMap(
+    value=10.0,
+    demand=10.0,
+    supply=10.0,
+    utilisation=0.5,
+    allocation=0.5,
+    consumption=_WarnValue(0.5, FutureWarning("")),
+    target=None,
+)
+
+
 class Logger(PoolDecorator):
     """
     Proxy which logs all changes of ``target.demand``
```

Comment:
I currently have a hard time understanding what the docstring is trying to say. Do you think it can be enhanced within this PR?